echidna/core/dsextract.py

```diff
@@ -0,0 +1,608 @@
+import numpy
+import rat
+import echidna.calc.constants as const
+
+
+def function_factory(dimension, **kwargs):
+    '''Factory function that returns a dsextract class
+    corresponding to the dimension (i.e. DS parameter)
+    that is being extracted.
+
+    Args:
+      dimension (str): to extract from a RAT DS/ntuple file.
+      \**kwargs (dict): to be passed to and checked by the extractor.
+
+    Raises:
+      IndexError: dimension is an unknown parameter
+
+    Retuns:
+      Extractor object.
+    '''
+    if dimension == "energy_mc":
+        return EnergyExtractMC(**kwargs)
+    elif dimension == "energy_reco":
+        return EnergyExtractReco(**kwargs)
+    elif dimension == "energy_truth":
+        return EnergyExtractTruth(**kwargs)
+    elif dimension == "radial_mc":
+        return RadialExtractMC(**kwargs)
+    elif dimension == "radial_reco":
+        return RadialExtractReco(**kwargs)
+    elif dimension == "radial3_mc":
+        return Radial3ExtractMC(**kwargs)
+    elif dimension == "radial3_reco":
+        return Radial3ExtractReco(**kwargs)
+    else:
+        raise IndexError("Unknown parameter: %s" % dimension)
+
+
+class Extractor(object):
+    '''Base class for extractor classes.
+
+    Args:
+      name (str): of the dimension
+      fid_r (float): Fiducial radius. Applies a cut to remove
```

Comment:
I'm not sure about the `fid_r` as a variable name. Was there a reason for choosing `fid_r` over something like `fv_radius` that we had before? You still use `av_radius` elsewhere. It's completely personal preference, but I just feel `fv_radius` is more explicit than `fid_r`.